src/demuxer.rs

```diff
@@ -333,18 +336,153 @@ mod tests {
     fn parse_headers() {
         let mut demuxer = MkvDemuxer::new();
 
-        let res = demuxer.parse_until_tracks(webm);
-        println!("got parsing res: {res:?}");
-        match res {
-            Ok((i, _)) => {
-                println!("offset: {} bytes", webm.offset(i));
-            }
-            e => {
-                println!("could not parse: {e:?}");
-            }
+        match demuxer.parse_until_tracks(webm) {
+            Ok((i, _)) => println!("parsed {} bytes", webm.offset(i)),
+            e => panic!("parse error: {e:?}"),
         }
 
-        println!("demuxer: {demuxer:#?}");
+        assert_eq!(
+            demuxer.header.expect("can parse EBML Header"),
```

Comment:
With `expect` shouldn't we use `cannot parse`? Same for other error messages.

Reply:
I followed [this advice](https://doc.rust-lang.org/std/result/enum.Result.html#recommended-message-style) from the standard library.

Reply:
Hmm, it seems different from the example immediately above, where the content of `expect` is before the error message, so it is positive for that reason I guess

Reply:
Oh, right. I'll change it real quick

Reply:
Oh wait, this can never be the case anyways. If `parse_until_tracks` (called a few lines earlier) returns `Ok`, then all these 4 fields will be `Some(...)`.

Even if they aren't, these are Options, not Results. There is no further error message. I'll just unwrap, should be fine/an obvious error.